### PR TITLE
Add concierge track events

### PR DESCRIPTION
**Pull request by unDemian**
## Summary
This PR adds analytics tracks for concierge events. Added the following events.
- calypso_concierge_book_calendar_step
- calypso_concierge_book_confirmation_step
- calypso_concierge_book_info_step
- calypso_concierge_cancel_step
- calypso_concierge_site_selection_step
- calypso_concierge_reschedule_calendar_step
- calypso_concierge_reschedule_confirmation_step
- calypso_concierge_appointment_cancellation_successful
- calypso_concierge_appointment_cancellation_error
- calypso_concierge_appointment_booking_successful
- calypso_concierge_appointment_booking_error
- calypso_concierge_appointment_rescheduling_successful
- calypso_concierge_appointment_rescheduling_error

## Testing
Code review only.

### CODE REVIEW

File: client/my-sites/people/people-invites/index.jsx [hunk 1]
```diff
@@ -37,14 +37,7 @@ class PeopleInvites extends React.PureComponent {
...
 			<Card key={ invite.invite_key }>
-				Invited
-				{ ' ' }
-				<Gravatar user={ gravatarUser } />
-				{ ' ' }
-				<strong>{ userNameOrEmail }</strong>
-				{ ' ' }
-				as
-				{ ' ' }
+				Invited <Gravatar user={ gravatarUser } /> <strong>{ userNameOrEmail }</strong> as{' '}
```
Comment (mattwondra) [inline client/my-sites/people/people-invites/index.jsx]: I'm not sure why this change was made but it might be worth reverting — the whitespace from before actually could be significant.

File: client/state/data-layer/wpcom/concierge/schedules/appointments/index.js [hunk 2]
```diff
@@ -27,6 +27,7 @@ import {
 	CONCIERGE_ERROR_NO_AVAILABLE_STAFF,
 } from 'me/concierge/constants';
 import fromApi from './from-api';
+import analytics from 'lib/analytics';
```
Comment (mattwondra) [inline client/state/data-layer/wpcom/concierge/schedules/appointments/index.js]: This line is actually what's causing tests to break. It's been a while since I've used `lib/analytics`, but as I recall simply importing it into a module causes tests to fail unless you mock it. So I'm guessing the failing tests are importing the whole `data-layer` so this import is causing them to fail.

The solution here is actually to use [the `state/analytics` helpers](https://github.com/Automattic/wp-calypso/blob/master/client/state/analytics/README.md) — all these events are directly tied to a dispatch, so you use `withAnalytics` to attach analytics metadata to the dispatched actions, which get picked up and handled in middleware.

That should stop the CI failures, and also be better long-term.

Comment (unDemian) [reply client/state/data-layer/wpcom/concierge/schedules/appointments/index.js]: Good catch, I updated the code to use state helpers.

File: client/my-sites/people/people-invites/index.jsx (hunk 1 above)
Comment (unDemian) [reply client/my-sites/people/people-invites/index.jsx]: Pretty weird because I did not touch this file, it might be some automatic update from Prettify or some other tool. I reverted the change.

**Comment by unDemian** [comment]
Thanks for taking the time, this is ready for re-review.

File: client/my-sites/people/people-invites/index.jsx (hunk 1 above)
Comment (mattwondra) [reply client/my-sites/people/people-invites/index.jsx]: This change showed back up. I think it's happening when you're merging master back in or something — that "touches" this file, and so Prettier auto-formats it.

Comment (unDemian) [reply client/my-sites/people/people-invites/index.jsx]: Prettier automatically re-formats the file, I guess when `/** @format */` was added 7 days ago to the file the pre-commit hook that formats the file was not trigger so the file remained not formatted.

The output looks the same https://cloudup.com/cbpRPfab9Q0 vs https://cloudup.com/cXRtcphgsWm so I guess we don't really need to revert it.
